embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/EmbraceDeliveryCacheManager.kt

```diff
@@ -184,29 +184,68 @@ internal class EmbraceDeliveryCacheManager(
         }
     }
 
-    override fun saveFailedApiCalls(failedApiCalls: DeliveryFailedApiCalls) {
+    /**
+     * Saves the [FailedApiCallsPerEndpoint] map to a file named [FAILED_API_CALLS_FILE_NAME].
+     */
+    override fun saveFailedApiCalls(failedApiCalls: FailedApiCallsPerEndpoint) {
         logger.logDeveloper(TAG, "Saving failed api calls")
         serializer.bytesFromPayload(
             failedApiCalls,
-            DeliveryFailedApiCalls::class.java
+            FailedApiCallsPerEndpoint::class.java
         )?.let {
             executorService.submit {
                 cacheService.cacheBytes(FAILED_API_CALLS_FILE_NAME, it)
             }
         }
     }
 
-    override fun loadFailedApiCalls(): DeliveryFailedApiCalls {
+    /**
+     * Loads the [FailedApiCallsPerEndpoint] map from a file named [FAILED_API_CALLS_FILE_NAME].
+     * If loadObject returns null, it tries to load the old version of the file which was storing
+     * [DeliveryFailedApiCalls] instead of [FailedApiCallsPerEndpoint].
+     */
+    override fun loadFailedApiCalls(): FailedApiCallsPerEndpoint {
         logger.logDeveloper(TAG, "Loading failed api calls")
-        val cached = executorService.submit<DeliveryFailedApiCalls> {
-            cacheService.loadObject(FAILED_API_CALLS_FILE_NAME, DeliveryFailedApiCalls::class.java)
+        val cached = executorService.submit<FailedApiCallsPerEndpoint> {
+            val loadFailedApiCalls = runCatching {
+                cacheService.loadObject(FAILED_API_CALLS_FILE_NAME, FailedApiCallsPerEndpoint::class.java)
+            }
+            if (loadFailedApiCalls.isSuccess) {
```

Comment:
We don't need to delete it because it will be overwritten the next time we save it. And deleting would be bad because we could be missing data if the requests retry don't succeed.